api/src/graphql/context.ts

```diff
@@ -28,11 +28,12 @@ export function createContext(params: CreateContextParams): Context {
   const { req } = params;
   const authorizationHeader = req.get('Authorization');
   const token = authorizationHeader?.replace('Bearer ', '');
+  const origin = req.get('Origin');
```

Comment:
@valiafetisov don't have prior xp on this one, googling did not yield me a definite answer.

Is relying on the origin header safe? can't it be forged? e.g. in tests i manage to put any origin i want and hence i imagine that this approach is not safe.

Reply:
The header can not be changed by the JavaScript running in the browser, but can be set in many other environments (like curl). It is sufficient for the issue

Reply:
Related question to @DeusAvalon. Is it enough to just `req.get('Origin')` on the application level to get the actual origin (not `frontend:3000`) or should ingress be additionally configured to pass proper header to the application? Or does it come via something like `x-origin`?

Reply:
Im pretty sure `req.get('Origin')` on Application level should give the correct Origin (actual Ingress URL) as the Ingress does not strip/replace any headers in our current configuration.

Reply:
Hmm, then what header contains internal `frontend:3000`? And what is [`proxy-add-original-uri-header`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#proxy-add-original-uri-header) for?

If possible, I would ask kirill to enable staging deployments on this branch to validate that it indeed works, but as the repo wasn't transferred yet, we would need to test (and potentially fix) it after this PR is merged, via our fork

Reply:
Hmm i see, yeah im not 100% sure either i would need to actually deploy a service and check which headers are being parsed to the service.

But the staging deployment could actually be done by me manually deploying it based on this branch. This way we can pre-validate if its actually working. If this would work for you i would organize this tmrw with @KirillDogadin-std together